packages/dd-trace/src/appsec/iast/vulnerability-reporter.js

```diff
@@ -94,20 +96,28 @@ function sendVulnerabilities (iastContext) {
   return IAST_JSON_TAG_KEY
 }
 
+function clearCache () {
+  VULNERABILITY_HASHES.clear()
+}
+
 function deduplicateVulnerabilities (vulnerabilities) {
-  const uniqueVulnerabilities = new Set()
-  return vulnerabilities.filter((vulnerability) => {
+  const deduplicated = vulnerabilities.filter((vulnerability) => {
     const key = `${vulnerability.type}${vulnerability.hash}`
-    if (!uniqueVulnerabilities.has(key)) {
-      uniqueVulnerabilities.add(key)
+    if (!VULNERABILITY_HASHES.has(key)) {
+      VULNERABILITY_HASHES.add(key)
       return true
     }
     return false
   })
+  if (VULNERABILITY_HASHES.size > VULNERABILITY_HASHES_MAX_SIZE) {
```

Comment:
Why cap it? And more importantly, why clear it completely when the max is reached?

Reply:
If the goal is to keep the most recent ones, why not use a LRU cache instead?

Reply:
It could be a good idea, I will talk about it with Java team, to maintain coherence in both tracers.